v3/src/components/data-display/models/data-configuration-model.ts

```diff
@@ -627,7 +627,7 @@ export const DataConfigurationModel = types
       // Invalidate cache when selection changes.
       addDisposer(self, reaction(
         () => self.dataset?.selection.values(),
-        () =>  self.allCasesForCategoryAreSelected.invalidateAll(),
+        () => self.clearCasesCache(),
```

Comment:
The problem with this solution is that `clearCasesCache` means that absolutely everything has to be recomputed from scratch, which is expensive, particularly given that selection changes are very frequent. At the very least, we should only do this when `displayOnlySelectedCases` is true. The goal is to invalidate what we need to invalidate for the current situation and no more.

Reply:
I've added a check for whether `displayOnlySelectedCases` is true. I also tried limiting the invalidations to some subset of what's in `clearCasesCache`, but the bug isn't fixed unless all of them are called. Does the current state seem good? Or should I spend more time trying to limit the invalidation somehow?